Applications/Spire/Source/Ui/TransitionView.cpp

```diff
@@ -0,0 +1,82 @@
+#include "Spire/Ui/TransitionView.hpp"
+#include <QLabel>
+#include <QMovie>
+#include <QTimer>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Ui/Box.hpp"
+#include "Spire/Ui/Layouts.hpp"
+
+using namespace Spire;
+using namespace Spire::Styles;
+
+TransitionView::TransitionView(QWidget* body, QWidget* parent)
+    : QWidget(parent),
+      m_body(body),
+      m_timer(new QTimer(this)),
+      m_status(Status::NONE) {
+  make_vbox_layout(this);
+  make_pending_widget();
+  m_timer->setSingleShot(true);
```

Comment:
This assumes that a widget is shown on construction, which is almost never the case.